docs/topics/external-cloud-provider.md

```diff
@@ -0,0 +1,29 @@
+# External Cloud Provider
+
+To deploy a cluster using [external cloud provider](https://github.com/kubernetes-sigs/cloud-provider-azure), create a cluster configuration with external cloud provider template. For example,
+
+```bash
+clusterctl config cluster ${CLUSTER_NAME} --from templates/cluster-template-external-cloud-provider.yaml --kubernetes-version v1.17.3 --control-plane-machine-count=3 --worker-machine-count=3 > ${CLUSTER_NAME}.yaml
```

Comment:
I'm hesitant to give instructions for using clusterctl with a local file here as users wouldn't necessarily have the repo cloned and devs wouldn't necessarily have the clusterctl binary... I think once we add the template a release it makes sense to give instructions on using the flavor, ie. `clusterctl config cluster ${CLUSTER_NAME} --flavor  external-cloud-provider --kubernetes-version v1.17.4 --control-plane-machine-count=3 --worker-machine-count=3 > ${CLUSTER_NAME}.yaml` but for devs there shouldn't be a requirement on clusterctl, they should just use kustomize build and apply the template. Or maybe we need to modify `make create-cluster` to allow overriding the `$TEMPLATE` or `$FLAVOR` value?

I realize this is not really actionable feedback for this PR but more questions that need to be answered as we figure out how kustomize/flavors should work together with @kubernetes-sigs/federation-wg 

So I guess my feedback for this PR is either:
1. Update the docs assuming the template flavor will exist in the upcoming release that includes this commit
`clusterctl config cluster ${CLUSTER_NAME} --flavor  external-cloud-provider --kubernetes-version v1.17.4 --control-plane-machine-count=3 --worker-machine-count=3 > ${CLUSTER_NAME}.yaml` (also changed k8s version to 1.17.4) , or
2.  Remove the clusterctl command and stay vague until we're able to update this with better instructions as part of the general "using flavors" upcoming PR:
`To deploy a cluster using [external cloud provider](https://github.com/kubernetes-sigs/cloud-provider-azure), create a cluster configuration with the [external cloud provider template](../../templates/cluster-template-external-cloud-provider.yaml).`

Reply:
Sounds good, updated with 2. We can always change it to `--flavor` after it lands in release.